server/src/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -1171,6 +1174,17 @@ public SnapshotInfo takeSnapshot(VolumeInfo volume) throws ResourceAllocationExc
         return snapshot;
     }
 
+    protected void backupSnapshotToSecondary(boolean asyncBackup, SnapshotStrategy snapshotStrategy, SnapshotInfo snapshotOnPrimary) {
```

Comment:
Would you mind documenting this method, and then creating unit test cases for it?